src/parser/core/modules/Statuses.js

```diff
@@ -0,0 +1,174 @@
+import ACTIONS from 'data/ACTIONS'
+import STATUSES from 'data/STATUSES'
+import Module from 'parser/core/Module'
+import {ItemGroup, Item} from './Timeline'
+import {getDataBy} from 'data'
+import React from 'react'
+
+const STATUS_APPLY_ON_PARTY_THRESHOLD_MILLISECONDS = 2 * 1000
+
+// Track statuses applied by actions
+export default class Statuses extends Module {
+	static handle = 'statuses'
+	static dependencies = [
+		'timeline',
+		'cooldowns',
+		'gcd',
+	]
+
+	static statusesStackMapping = {}
+
+	_statuses = {}
+	_groups = {}
+	_statusToActionMap = {}
+	_actionToMergeNameMap = {}
+
+	constructor(...args) {
+		super(...args)
+
+		const ids = [this.parser.player.id, ...this.parser.player.pets]
```

Comment:
medium; is this intentional? `player.pets` is an array of pet objects, not IDs. If you want just actor IDs, it'd be `[this.parser.player.id, ...this.parser.player.pets.map(pet => pet.id)]`

Reply:
done